### PR TITLE
Read PKey and Transport Mode from parent device

### PR DESCRIPTION
**Pull request by ThomasGebert**
If PKey is different from the parent device, the virtual device in the container will have LOWERLAYERDOWN.
Also setting Transport Mode to the same as the parent device seems to be sane.

### CODE REVIEW

**Comment by adrianchiris** [comment]
@e0ne PTAL

overall this change looks OK to me.

File: pkg/ipoib/ipoib.go [hunk 1]
```diff
@@ -100,15 +101,40 @@ func (im *ipoibManager) CreateIpoibLink(conf *types.NetConf, ifName string, netn
...
 	ipoibLink := &netlink.IPoIB{
 		LinkAttrs: netlink.LinkAttrs{
 			Name:        tmpName,
 			ParentIndex: m.Attrs().Index,
 			// Due to kernal bug create the link then move it to the desired namespace
 			//		Namespace:   netlink.NsFd(int(curNetns.Fd())),
 		},
-		Pkey:   0x7fff,
-		Mode:   netlink.IPOIB_MODE_DATAGRAM,
+		//Pkey:   0x7fff,
```
Comment (adrianchiris) [inline pkg/ipoib/ipoib.go:134]: please remove comments

**Comment by adrianchiris** [comment]
Thanks @ThomasGebert  ! appreciate the contribution

could you share how you are using ipoib cni ?

in your deployment you have the master interface with a pkey other than the default one (management pkey) ?

**Comment by adrianchiris** [comment]
also appologies on the late review, i must have missed the notification for this PR

File: pkg/ipoib/ipoib.go (hunk 1 above)
Comment (ThomasGebert) [reply pkg/ipoib/ipoib.go]: Just removed the comments, which referenced the origin settings.

**Comment by ThomasGebert** [comment]
> Thanks @ThomasGebert ! appreciate the contribution
> 
> could you share how you are using ipoib cni ?
> 
> in your deployment you have the master interface with a pkey other than the default one (management pkey) ?

@adrianchiris : the reason for this PR has its source in using Intel Omnipath as underlaying fabric which does not use _0x7fff_ as default PKey, I can't remember right now which one is used there by default. So I thought it was a good idea to get the values used here from the master device and apply it to the virtual Kubernetes network device.

**Comment by adrianchiris** [comment]
OK makes sense. We need to test this one with Mellanox NIC to make sure this change plays nice with that as well b4 merging.

**Comment by ThomasGebert** [comment]
@adrianchiris : I have added the origin defaults again for PKey and Mode if the file in _sysfs_ are not readable. 
Not sure if that is the desired behaviour?

**Comment by adrianchiris** [comment]
@ThomasGebert Please take a look at PR #35  if this works for you, we can merge it.

**Comment by ThomasGebert** [comment]
> @ThomasGebert Please take a look at PR #35 if this works for you, we can merge it.

@adrianchiris : your PR #35 looks good and is a neat solution. Unfortunately I do not have access to the Omnipath fabric anymore and cannot test it. 

Should I just close this PR?

**Comment by adrianchiris** [comment]
Ack, yes, please close this one. if there are additional changes for omni path lets discuss in separate PR/Issue

**Pull request closed without merging**